tests/src/main/java/net/neoforged/neoforge/debug/capabilities/VanillaHandlersTests.java

```diff
@@ -78,6 +80,36 @@ public static void testComposterInvalidation(ExtendedGameTestHelper helper) {
         helper.succeed();
     }
 
+    @GameTest
+    @EmptyTemplate
+    @TestHolder(description = "Tests that non-compostable items cannot be inserted into a composter")
+    public static void testComposterCannotAcceptNonCompostables(ExtendedGameTestHelper helper) {
+        var composterPos = new BlockPos(1, 1, 1);
+
+        helper.setBlock(composterPos, Blocks.COMPOSTER.defaultBlockState());
+
+        var nonCompostable = new ItemStack(Blocks.BARRIER, 1);
+        if (ComposterBlock.getValue(nonCompostable) > 0)
+            helper.fail("Assumption failed: expected " + nonCompostable + " to be non-compostable");
+
+        // Of particular note to be tested here is the 'null' side; see #2572
+        // "IItemHandler for null side of Composter allows items without compost value to be inserted"
+        var sides = new Direction[] { null, Direction.UP, Direction.DOWN, Direction.NORTH, Direction.EAST, Direction.SOUTH, Direction.WEST };
+
+        for (Direction side : sides) {
+            var capability = helper.getLevel().getCapability(Capabilities.ItemHandler.BLOCK, helper.absolutePos(composterPos), side);
+            if (capability == null)
+                helper.fail("Expected composter to have a block item handler capability for side " + side);
+
+            assert capability != null; // The helper#fail call above ensures this is not null
```

Comment:
This is weird. We have a `helper.requireCapability` method which will make this unnecessary I believe.

Reply:
Ah, I was unaware that method existed. The rest of the class doesn't mention it, and I didn't look at other classes.